### PR TITLE
Support removal of EOL semicolons;

### PR DESCRIPTION
**Pull request by joshhartigan**

### CODE REVIEW

**Comment by joshhartigan** [comment]
apologies for the 3 commits  rather than 1, i made a few mistakes

File: test.js [hunk 1]
```diff
@@ -24,4 +24,7 @@ function foo()  {}
 
 function(){}
 function foo (){}
-function foo ()  {}
\ No newline at end of file
+function foo ()  {}
+
+console.log('hello, world!');
```
Comment (yoshuawuyts) [inline test.js]: not sure this was intentional

Comment (yoshuawuyts) [inline test.js]: Not sure this was intentional

File: test.js [hunk 2]
```diff
@@ -24,4 +24,7 @@ function foo()  {}
...
+
```
Comment (yoshuawuyts) [inline test.js]: Not sure this was intentional

Comment (juliangruber) [inline test.js]: one could remove the newline, but the console log tests the eol change this pr introduces

Comment (yoshuawuyts) [inline test.js]: ah, I see. :) Was a bit confused because of the words being printed; `'correct EOL'` or similar might be a bit more clear.

**Comment by jb55** [comment]
> apologies for the 3 commits rather than 1, i made a few mistakes

If you're concerned about this stuff you can always `git rebase -i master` and mark the extra commits with `fixup`

**Comment by jb55** [comment]
Thanks!

**Pull request merged**